src/coreclr/tools/Common/JitInterface/ThunkGenerator/InstructionSetGenerator.cs

```diff
@@ -385,9 +377,17 @@ namespace Internal.JitInterface
     public enum InstructionSet
     {
         ILLEGAL = 0,
-        NONE = 63,
 ");
 
+            int maxJitNames = 0;
+            foreach (string architecture in _architectures)
+            {
+                maxJitNames = (_architectureJitNames[architecture].Count) > maxJitNames ? (_architectureJitNames[architecture].Count) : maxJitNames;
+            }
+            maxJitNames++;
+
+            tr.WriteLine($"        NONE = {maxJitNames},");
+
```

Comment:
If we went with the `FlagsFieldCount` approach, this could just be
```cpp
        ILLEGAL = 0,
        NONE = ");

        tr.Write((FlagsFieldCount * sizeof(ulong) * 8) - 1);

        tr.Write(@",
");
```

Reply:
Yes. It's a question of using a hardcoded `FlagsFieldCount ` value vs using a 'calculated value. If you think this works better, I'm okay with adding a private field `FlagsFieldCount ` and using it.

Reply:
I personally prefer the `FlagsFieldCount` approach because it gives us a "update this one line" fix whenever we need to grow things.

It also becomes very explicit that the size is growing which should make it easier to handle versioning over time.

Reply:
I made the change and switched to `FlagsFieldCount`